src/insights_client/tests/test_metrics.py

```diff
@@ -34,27 +35,92 @@ def factory(config):
 @fixture
 def rhsm_config_file_factory():
     def factory(**config):
+        # keep these default values initialized
+        if "hostname" not in config:
+            config["hostname"] = "subscription.rhsm.redhat.com"
+        if "port" not in config:
+            config["port"] = "443"
         config = u"""[server]
-hostname = cert-api.access.redhat.com
-port = 443
-proxy_hostname = %s
-proxy_port = %s
-proxy_user = %s
-proxy_password = %s
-
+%s
+%s
+%s
+%s
+%s
+%s
 [rhsm]
-repo_ca_cert =
-consumerCertDir =
-""" % (
-            config.get("proxy_hostname", ""),
-            config.get("proxy_port", ""),
-            config.get("proxy_user", ""),
-            config.get("proxy_password", ""),
-        )
+%s
+%s
+""" % tuple(
+            "%s = %s" % (key, config[key]) if key in config else ""
+            for key in ("hostname", "port", "proxy_hostname", "proxy_port", "proxy_user", "proxy_password", "repo_ca_cert", "consumerCertDir"))
         return _tempfile(config)
 
     return factory
 
+@patch("insights_client.metrics._proxy_settings")
+def test_http_client_init_missing_server_section(proxy_settings, config_file_factory):
+    '''
+    Verify that when the "server" section is missing, RHSM configuration is ignored
+    and insights-client configuration is preferred
+    '''
+    config_file = config_file_factory("username=testuser\npassword=testpass\nauthmethod=BASIC")
+    rhsm_config_file = _tempfile("[nonsense]\ntest=test")
+    metrics_client = MetricsHTTPClient(config_file=config_file.name, rhsm_config_file=rhsm_config_file.name)
+    assert metrics_client.auth == ("testuser", "testpass")
+    assert not metrics_client.cert
```

Comment:
We may also want to check that _metrics_client.verify_ is not set.

```suggestion
    assert not metrics_client.cert
    assert not metrics_client.verify
```

Reply:
In this case, `Session().verify` is default to true, but I've now added a test for the value regardless.

Reply:
Oh.